include/range/v3/view/single.hpp

```diff
@@ -38,11 +39,11 @@ namespace ranges
         {
         private:
             friend struct range_access;
-            Val value_;
+            semiregular_t<Val> value_;
             struct cursor
             {
             private:
-                Val value_;
+                semiregular_t<Val> value_;
```

Comment:
`cursor` needs `value_` to have a `Semiregular` type in order to be `Semiregular` itself.